service/header/subscription_test.go

```diff
@@ -54,9 +51,16 @@ func TestSubscriber(t *testing.T) {
 	err = p2pSub2.Start(context.Background())
 	require.NoError(t, err)
 
+	err = net.ConnectAllButSelf()
+	time.Sleep(400 * time.Millisecond) // wait for peers to finish their connection.
```

Comment:
Is this necessary even though we wait for the method to return?

Reply:
If it is necessary, is there a way to learn about peers finishing up their conns reliably independent of a sleep? If not, then maybe even increase those milliseconds as this might fail on time to time if there is load on the github servers.

Reply:
We had sleeps before. They do not fix the issue unless the sleep is ridiculously big.

Reply:
I have removed the need for a sleep by watching the peer store of both peers, and making sure that both have connected. I run the check against a timeout of 2 seconds to acknowledge the wonkiness of github actions.

I have run the test in a loop 1000 times to make sure it was a good fix, and it held well.